### PR TITLE
DTS: Fix irq aliases generation

### PR DESCRIPTION
**Pull request by galak**
We generate the following today:
   DT_NXP_KINETIS_GPIO_GPIO_D_IRQ
however it should have been:
   DT_NXP_KINETIS_GPIO_GPIO_D_IRQ_0

This PR fixes this up and deprecates the old defines.

### CODE REVIEW

File: scripts/dts/extract/interrupts.py [hunk 1]
```diff
@@ -69,10 +69,27 @@ def extract(self, node_path, prop, names, def_label):
...
+                            full_name,
+                            prop_alias)
+                    else:
+                        add_prop_aliases(
+                            node_path,
+                            lambda alias:
+                                '_'.join([str_to_label(alias)] +
+                                         l_cell_prefix + name + l_cell_name),
+                            full_name,
+                            prop_alias)
```
Comment (ulfalizer) [inline scripts/dts/extract/interrupts.py]: The `if` and `else` blocks seem to be identical.

**Comment by ulfalizer** [comment]
Looks nice to me besides that `if`/`else` thing, as far as I understand it.

When there can be one or more of something, I think we should always add `_0`. That streamlines things. Probably goes for other places too.

File: scripts/dts/extract/interrupts.py (hunk 1 above)
Comment (galak) [reply scripts/dts/extract/interrupts.py]: They are, its so in a follow on commit we'll mark one of them deprecated.

File: scripts/dts/extract/interrupts.py [hunk 2]
```diff
@@ -69,10 +69,27 @@ def extract(self, node_path, prop, names, def_label):
...
+                    else:
+                        add_prop_aliases(
+                            node_path,
+                            lambda alias:
+                                '_'.join([str_to_label(alias)] +
+                                         l_cell_prefix + name + l_cell_name),
+                            full_name,
+                            prop_alias, True)
```
Comment (ulfalizer) [inline scripts/dts/extract/interrupts.py]: Could do this to get rid of the `if/else`:

 ```
 add_prop_aliases(
    node_path,
    lambda alias:
        '_'.join([str_to_label(alias)] +
                 l_cell_prefix + name + l_cell_name),
    full_name,
    prop_alias,
    not name)
```

**Review by ulfalizer — APPROVED** [review]

**Pull request merged**